experiments/type-system/src/main.rs

```diff
@@ -0,0 +1,95 @@
+#![allow(dead_code)]
+
+mod ast;
+mod error;
+mod lexer;
+mod parser;
+
+use std::path::PathBuf;
+
+use clap::Parser;
+
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct Args {
+  #[clap(value_parser)]
+  input: PathBuf,
+  #[clap(value_parser)]
+  output: PathBuf,
+}
+
+struct Symbols;
+
+impl ast2str::Symbols for Symbols {
```

Comment:
There's a built-in thing for this, it's called TestSymbols iirc. Check the parser branch.

Reply:
I tried using `https://insta.rs/` for the snapshot testing, and it's really nice, but it only supported serde as a way to serialize (if I didn't want to use unstructured strings), so I ended up removing the `AstToStr` dependency here

Reply:
That's fine, but note that insta has [`assert_debug_snapshot`](https://docs.rs/insta/latest/insta/macro.assert_debug_snapshot.html#), and ast2str has [`DisplayAsDebugWrapper`](https://docs.rs/ast2str-lib/1.2.1/ast2str_lib/utils/index.html#example)